### PR TITLE
Adds `PopupMenuPosition position` to the `PopupMenuThemeData`

### LINKED ISSUE #110264: Could `PopupMenuPosition position` from `PopupMenuButton` be added to the `PopupMenuThemeData` ?
## Use case

I'm using `PopupMenuButton` in my app, and for all of them, I have to specify `position: PopupMenuPosition.under` everywhere I used it. It would be useful to be able to set it in the theme instead.

## Proposal

Add an entry `PopupMenuPosition position` in the `PopupMenuThemeData`.

### PR DESCRIPTION
**Pull request by ValentinVignal**
Closes https://github.com/flutter/flutter/issues/110264

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by adilwahla — COMMENTED** [review]
Pretty cool, i think should be merged after reviews by team

**Comment by ValentinVignal** [comment]
Hello @gspencergoog, is there something more I should do for this pull request?

File: packages/flutter/test/material/popup_menu_theme_test.dart [hunk 1]
```diff
@@ -78,16 +82,21 @@ void main() {
...
-              itemBuilder: (BuildContext context) {
-                return <PopupMenuEntry<void>>[
-                  PopupMenuItem<void>(
-                    key: popupItemKey,
-                    child: const Text('Example'),
-                  ),
-                ];
-              },
+            Padding(
+              // The padding makes sure the menu as enough space to around to
```
Comment (gspencergoog) [inline packages/flutter/test/material/popup_menu_theme_test.dart:86]:
```suggestion
              // The padding makes sure the menu has enough space around it to
```

File: packages/flutter/test/material/popup_menu_theme_test.dart [hunk 2]
```diff
@@ -202,20 +224,26 @@ void main() {
...
-                return <PopupMenuEntry<void>>[
-                  PopupMenuItem<void>(
-                    key: popupItemKey,
-                    textStyle: textStyle,
-                    child: const Text('Example'),
-                  ),
-                ];
-              },
+            Padding(
+              // The padding makes sure the menu as enough space to around to
```
Comment (gspencergoog) [inline packages/flutter/test/material/popup_menu_theme_test.dart:228]:
```suggestion
              // The padding makes sure the menu has enough space around it to
```

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Sorry for the delay.  This looks fine, except for some minor comment nits.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/test/material/popup_menu_theme_test.dart (hunk 1 above)
Comment (ValentinVignal) [reply packages/flutter/test/material/popup_menu_theme_test.dart:86]: Thank you, I created a new PR to fix this https://github.com/flutter/flutter/pull/111060

**Pull request merged**